pdns_api/config.go

```diff
@@ -1,35 +1,66 @@
 package pdns_api
 
 import (
-	"github.com/BurntSushi/toml"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/viper"
 )
 
 const AuthTypeHTTP = "http"
 const AuthTypeToken = "token"
 
+func BindEnvs(iface interface{}, parts ...string) {
+	ifv := reflect.ValueOf(iface)
+	ift := reflect.TypeOf(iface)
+	for i := 0; i < ift.NumField(); i++ {
+		v := ifv.Field(i)
+		t := ift.Field(i)
+		tv, ok := t.Tag.Lookup("mapstructure")
+		if !ok {
+			continue
+		}
+		switch v.Kind() {
+		case reflect.Struct:
+			BindEnvs(v.Interface(), append(parts, tv)...)
+		default:
+			viper.BindEnv(strings.Join(append(parts, tv), "."))
+		}
+	}
+}
 func NewConfig(confPath string) (Config, error) {
 	var conf Config
 	defaultConfig(&conf)
+	viper.SetConfigFile(confPath)
+
+	viper.AutomaticEnv()
+	viper.SetEnvPrefix("PIR5")
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+
+	if err := viper.ReadInConfig(); err != nil {
+		return conf, err
+	}
 
-	if _, err := toml.DecodeFile(confPath, &conf); err != nil {
+	BindEnvs(conf)
```

Comment:
If this is not the case
```toml
user = "user"
# pasword = "nothing"
```
In this case of the above configuration, the environment variable `PIR5_PASSWORD` cannot be used.